### PR TITLE
Add heredoc

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: autoload/vimlparser.vim [hunk 1]
```diff
@@ -407,6 +409,7 @@ endfunction
 " CURLYNAMEPART .value
 " CURLYNAMEEXPR .value
 " LAMBDA .rlist .left
+" HEREDOC .rlist .op .value
```
Comment (mattn) [inline autoload/vimlparser.vim]: HEREDOC .rlist .op .str

File: autoload/vimlparser.vim [hunk 2]
```diff
@@ -1522,6 +1567,13 @@ function! s:VimLParser.parse_cmd_let()
   if s2 ==# '+=' || s2 ==# '-=' || s2 ==# '.=' || s2 ==# '..=' || s2 ==# '*=' || s2 ==# '/=' || s2 ==# '%='
     call self.reader.getn(len(s2))
     let node.op = s2
+  elseif s2 ==# '=<<'
+    call self.reader.getn(len(s2))
+    call self.reader.skip_white()
+    let node.op = '='
```
Comment (tyru) [inline autoload/vimlparser.vim]: why not `=<<` ?

File: autoload/vimlparser.vim [hunk 3]
```diff
@@ -1483,6 +1486,46 @@ function! s:VimLParser.parse_cmd_call()
...
+    let line = self.reader.getn(-1)
+    if line ==# node.op
+      let node.str = join(lines, "\n") . "\n"
+      return node
+    endif
+    call add(lines, line)
+    call self.reader.get()
+  endwhile
+  return s:NIL
+endfunction
```
Comment (tyru) [inline autoload/vimlparser.vim]: is there `trim` support?

```vim
let text =<< trim END
   if ok
     echo 'done'
   endif
END
```

File: autoload/vimlparser.vim (hunk 2 above)
Comment (mattn) [reply autoload/vimlparser.vim]: done

File: autoload/vimlparser.vim (hunk 3 above)
Comment (mattn) [reply autoload/vimlparser.vim]: Yes, supported.

**Comment by mattn** [comment]
PTAL

**Comment by tyru** [comment]
🙌

**Comment by tyru** [comment]
@mattn please tell me if it's available to review 🙆‍♂

**Comment by tyru** [comment]
Current heredoc implementations may have more problems than we found.

https://github.com/vim-jp/issues/issues/1294
https://github.com/vim/vim/pull/4713

So we can skip heredoc support now, we should work on other issues.
(e.g. review & merge currently opening pull requests)

**Comment by tyru** [comment]
@mattn this can be merged if you fix conflict 👍

**Comment by tyru** [comment]
I forgot I have permission for this branch. fixed remaining vint errors.
I think this branch is mergeable. merge it

**Pull request merged**